src/components/radioGroup/radio/radio.tsx

```diff
@@ -0,0 +1,64 @@
+import * as RadioGroup from '@radix-ui/react-radio-group';
+import classNames from 'classnames';
+import type React from 'react';
+import { useId, type ComponentPropsWithoutRef } from 'react';
+import { Icon, IconType } from '../../icon';
+
+export interface IRadioProps extends ComponentPropsWithoutRef<'button'> {
+    /**
+     * Radio label
+     */
+    label: string;
+    /**
+     * Indicates whether the radio input is disabled
+     */
+    disabled?: boolean;
+    /**
+     * Value of the radio input
+     */
+    value: string;
+    /**
+     * Variant indicating the label positioning
+     * @default right
+     */
+    variant?: 'left' | 'right';
+}
+
+/**
+ * Input Radio component
+ *
+ * **NOTE**: The component must be used inside a `<RadioGroup />` component in order to work properly.
+ */
+export const Radio: React.FC<IRadioProps> = ({ label, variant = 'right', id, value, className, ...rest }) => {
+    const randomId = useId();
+    const processedId = id ?? randomId;
+
+    const itemClasses = classNames(
+        'group peer rounded-full outline-none', // default
+        'focus:outline-none focus-visible:ring focus-visible:ring-primary focus-visible:ring-offset', // focus
+        { 'order-2': variant === 'left' },
+    );
+
+    const labelClasses = classNames(
+        'text-base leading-tight text-neutral-500', // default
+        'hover:text-neutral-800', // hover
+        'peer-disabled:peer-data-[state=unchecked]:text-neutral-300', // unchecked and disabled
+    );
+
+    return (
+        <div className={classNames('flex items-center gap-2 rounded px-0.5 md:gap-3', className)}>
```

Comment:
I'd suggest to use a padding left/right on the label instead of a gap so that the space between the indicator and label is also clickable as most of other libraries do. Also other libraries mark the label as clickable

Reply:
Oooh nice catch 😄